pkg/apis/proto/function/v1/udfunction.proto

```diff
@@ -44,6 +44,8 @@ message Datum {
   bytes value = 2;
   EventTime event_time = 3;
   Watermark watermark = 4;
+  string id = 5;
+  uint64 num_delivered = 6;
```

Comment:
should we create a new type and embed it in Datum?

Reply:
I'm fine with both. However I do see wrapping them in a new type such as `Metadata` makes it more clear.

Reply:
Updated.